### PR TITLE
Add screen for installation instructions

### PR DESCRIPTION
**Pull request by lingyili**

### CODE REVIEW

**Comment by baodvu** [comment]
Screen doesn't display properly in landscape mode. Please fix. Make sure you use Auto Layout.

screen shot 2016-08-31 at 2 13 31 pm

File: MagicSpell/InstrPageViewController.swift [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        labelForTitle = UILabel()
+        labelForTitle.text = "MagicSpell"
+        labelForTitle.font = UIFont.systemFontOfSize(50)
+        labelForTitle.sizeToFit()
+        labelForTitle.left = UIScreen.mainScreen().bounds.size.width/2 - labelForTitle.width/2
+        labelForTitle.top = 50
+        self.view.addSubview(labelForTitle)
+        
+        labelForSubtitle = UILabel()
+        labelForSubtitle.text = "To SetUp MagicSpell Keyboard, follow these steps:"
```
Comment (baodvu) [inline MagicSpell/InstrPageViewController.swift]: s/SetUp/set up

File: MagicSpell/InstrPageViewController.swift [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+class InstrPageViewController: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    
+    var tableViewForInstruction: UITableView!
+    var cellForInstruction : InstrPageTableViewCell!
+    var labelForTitle : UILabel!
+    var labelForSubtitle : UILabel!
+    var btnForAlreadyDone : UIButton!
+    
+    var instructionSet = ["1.  Open your device settings", "2.  General", "3.  Keyboard", "4.  Keyboards", "5.  Add new keyboard", "6.  MagicSpell keyboard", "7.  Tap MagicSpell", "8.  Allow Full access"]
```
Comment (baodvu) [inline MagicSpell/InstrPageViewController.swift]: s/settings/Settings
s/Add new keyboard/Add New Keyboard
s/MagicSpell keyboard/MagicSpell
s/Tap MagicSpell/Tap "MagicSpellKeyboard -- MagicSpell"
Remove "Allow Full Access" -  we don't need special permissions for the keyboard.

File: MagicSpell/MagicSpell-Bridging-Header.h [hunk 3]
```diff
@@ -0,0 +1,5 @@
+//
+//  Use this file to import your target's public headers that you would like to expose to Swift.
+//
+
+#import "UIView+Layout.h"
```
Comment (baodvu) [inline MagicSpell/MagicSpell-Bridging-Header.h]: new line at EOF

**Comment by baodvu** [comment]
Build warning: MagicSpell/Base.lproj/Main.storyboard: Segues initiated directly from view controllers must have an identifier
Please fix

File: MagicSpell/MacroFunction.swift [hunk 4]
```diff
@@ -0,0 +1,17 @@
+//
+//  MacroFunction.swift
```
Comment (baodvu) [inline MagicSpell/MacroFunction.swift]: It's not a macro. Please rename to something like HexColor.swift

File: MagicSpell/UIView+Layout.m [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+    frame.size = size;
+    self.frame = frame;
+}
+
+-(CGPoint)origin
+{
+    return self.frame.origin;
+}
+
+-(void)setOrigin:(CGPoint)origin
```
Comment (baodvu) [inline MagicSpell/UIView+Layout.m]: There are lots of unnecessary functions here, like this one (setOrigin). Please remove.

File: MagicSpell/MacroFunction.swift (hunk 4 above)
Comment (lingyili) [reply MagicSpell/MacroFunction.swift]: maybe we need more helper function later beside hexcolor

File: MagicSpell/UIView+Layout.m (hunk 5 above)
Comment (lingyili) [reply MagicSpell/UIView+Layout.m]: the purpose of these functions is to skip frame

Comment (baodvu) [reply MagicSpell/UIView+Layout.m]: I tried removing origin and setOrigin and it still worked.

File: MagicSpell/MacroFunction.swift (hunk 4 above)
Comment (baodvu) [reply MagicSpell/MacroFunction.swift]: I don't see any further code will be needed. In the event that we do need to add more helper functions, we can put them in another file if needed. "Functions.swift" tells me nothing about what this file does.

File: MagicSpell/UIView+Layout.m (hunk 5 above)
Comment (lingyili) [reply MagicSpell/UIView+Layout.m]: did you also remove origin from the header file?

**Comment by baodvu** [comment]
lgtm

[Approved with PullApprove](https://pullapprove.com/baodvu/MagicSpell/pull-request/32/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=baodvu/MagicSpell)

**Pull request merged**